File: Leetcode/476/476.cpp
constexpr int MAXN = 32;

class Solution {
public:
    int findComplement(int num) {
        bitset<MAXN> A(num);
        auto s = A.flip().to_string();
        deque<char> tempDeque (s.begin(), s.end());
        while (tempDeque.size() > 1 && tempDeque.front() == '1') {
            tempDeque.pop_front();
        }
        s.clear();
        while (!tempDeque.empty()) {
            s += tempDeque.front();
            tempDeque.pop_front();
        }
        return bitset<MAXN>(s).to_ulong();
    }
};

File: Leetcode/476/476.go
func findComplement(num int) int {
	var a []int
	for num > 0 {
		if num&1 != 0 {
			a = append(a, 1)
		} else {
			a = append(a, 0)
		}
		num >>= 1
	}
	for i := 0; i < len(a); i++ {
		a[i] = 1 - a[i]
	}
	ans := 0
	val := 1
    for i := 0; i < len(a); i++ {
		ans += val * a[i]
		val <<= 1
	}
	return ans
}